Fix HashSet example code

bgrainger opened the pull request:
Use `StringComparer.OrdinalIgnoreCase` (a built-in type) instead of providing a buggy custom hash code implementation.

bgrainger commented:
Fixes #4291.

AntonLapounov commented on samples/snippets/csharp/VS_Snippets_CLR_System/system.Collections.Generic.HashSet_ExceptWith/cs/source2.cs:
```diff
@@ -80,13 +80,13 @@ class SameVehicleComparer : EqualityComparer<string>
     public override bool Equals(string s1, string s2)
     {
-        return s1.Equals(s2, StringComparison.CurrentCultureIgnoreCase);
+        return string.Equals(s1, s2, StringComparison.OrdinalIgnoreCase);
```
Wouldn't `StringComparer.OrdinalIgnoreCase.Equals(s1, s2)` be simpler and more logical here?  This class basically duplicates StringComparer.OrdinalIgnoreCase.

bgrainger replied:
That's an excellent point; I'm not sure if the documentation is trying to show *how* to write an equality comparer, or simply how to use a custom one. I think showing how to use `StringComparer.OrdinalIgnoreCase` is the better approach; will push an update.

bgrainger committed:
Remove SameVehicleComparer.

It had implementation bugs and StringComparer.OrdinalIgnoreCase already
solves this problem.

rpetrusha reviewed (approved):
Thank you so much for your work in improving this example, @bgrainger! We really appreciate it. We'll merge your PR now. The new version of the code will appear on docs.microsoft.com in the next day or two.
